### ISSUE TITLE
Null reference when Publish to folder for solution selected

### ISSUE THREAD

**Issue by mrward** [mono/monodevelop#6468]
VS Mac: 7.7.0.1750
Git revision: 743007e584bcc371b4f8314c199cfce858ebc7da
Build date: 2018-10-24 17:03:24+00
Build branch: release-7.7
Xamarin extensions: 4eb7bc2cc6d74dadc5b397d8c5c13f334bc258b3

1. Create a new .NET Core console project.
2. Right click the solution and select Publish - Publish to Folder.
3. Dialog is not displayed. IDE log has error:

```
Failed to publish project
System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.AspNetCore.Dialogs.PublishToFolderDialog.get_BinBaseUri () [0x00000] in /Users/vsts/agent/2.141.1/work/1/s/monodevelop/main/src/addins/MonoDevelop.AspNetCore/MonoDevelop.AspNetCore.Dialogs/PublishToFolderDialog.cs:26 
  at MonoDevelop.AspNetCore.Dialogs.PublishToFolderDialog.Initialize () [0x00131] in /Users/vsts/agent/2.141.1/work/1/s/monodevelop/main/src/addins/MonoDevelop.AspNetCore/MonoDevelop.AspNetCore.Dialogs/PublishToFolderDialog.cs:69 
  at MonoDevelop.AspNetCore.Dialogs.PublishToFolderDialog..ctor (MonoDevelop.AspNetCore.Commands.PublishCommandItem publishCommandItem) [0x0000d] in /Users/vsts/agent/2.141.1/work/1/s/monodevelop/main/src/addins/MonoDevelop.AspNetCore/MonoDevelop.AspNetCore.Dialogs/PublishToFolderDialog.cs:35 
  at (wrapper remoting-invoke-with-check) MonoDevelop.AspNetCore.Dialogs.PublishToFolderDialog..ctor(MonoDevelop.AspNetCore.Commands.PublishCommandItem)
  at MonoDevelop.AspNetCore.Commands.PublishToFolderBaseCommandHandler.Run (System.Object dataItem) [0x00026] in /Users/vsts/agent/2.141.1/work/1/s/monodevelop/main/src/addins/MonoDevelop.AspNetCore/MonoDevelop.AspNetCore.Commands/PublishToFolderBaseCommandHandler.cs:30 

```

Possibly this could be fixed by only enabling the Publish menu when the project node is selected.

> VS bug [#715111](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/715111)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.8.0.1038 (master)

Pull Request #6466 merged by: Jose Miguel Torres
Author: mono
Commit: 0bc0954ad91c133efb28ef8d2b1269c0638dadbd (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.1776 (release-7.7)

Pull Request #6473 merged by: Lluis Sanchez
Author: mono
Commit: d387b70696545900297d9e115240fa7a4fadbec7 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.509 (release-8.0)

Pull Request #6466 merged by: Jose Miguel Torres
Author: mono
Commit: 0bc0954ad91c133efb28ef8d2b1269c0638dadbd (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #6466 merged by: Jose Miguel Torres
Author: mono
Commit: 0bc0954ad91c133efb28ef8d2b1269c0638dadbd (xamarin/monodevelop)
